### PR TITLE
Added a WKWebView based example

### PR DESCRIPTION
**Pull request by dodikk**
WKWebView does not need this POD to achieve the same thing.
Still, I think the example and a readme mention might be useful.

Based on https://github.com/jivesoftware/JiveAuthenticatingHTTPProtocol/pull/9
After merging that one, the diffs should become more brief.

### CODE REVIEW

**Comment by dodikk** [comment]
The "foo/bar" example with basic authorization works fine in WKWebView.
However, my "video with authorization" example fails (while it works fine with `UIWebView` and this library).

I'll push more commits when I discover the correct settings for `WKWebView`.

**Comment by dodikk** [comment]
Using an isolated non-persistent site data store pops up the alert for NTLM but that does not help the playback yet.

```obj-c
WKWebViewConfiguration* config = [WKWebViewConfiguration new];
    config.allowsInlineMediaPlayback = YES;
    config.allowsPictureInPictureMediaPlayback = NO;
    config.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeAll;
    config.websiteDataStore = [WKWebsiteDataStore nonPersistentDataStore];
```

**Comment by dodikk** [comment]
> Using an isolated non-persistent site data store pops up the alert for NTLM but that does not help the playback yet.

Never mind, the video's URL scheme was `http` which is "banned" by default due to Apple security policy. So that the exceptions should be added explicitly.

Alternatively, JavaScript injection could be used if the server can serve videos via `https`.  For example, https://gist.github.com/dodikk/e028c03cd809271c950c135a3abd4240

The `config.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeAll;` and `Deployment Target >= iOS 10.0` are impactful as well. Without those options the legacy project did not play the video while the prototype from this pull request did work.

File: JiveAuthenticatingHTTPProtocolDemo/WKWebViewDemo/ViewController.m [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    self.webView.navigationDelegate = self;
+        
+    NSString* txtUrl = @"https://httpbin.org/basic-auth/foo/bar";
+    NSURL* url = [NSURL URLWithString: txtUrl];
+    NSURLRequest* request = [[NSURLRequest alloc] initWithURL: url];
+    [self.webView loadRequest: request];
+}
+
+#pragma mark - JAHPAuthenticatingHTTPProtocolDelegate
```
Comment (hborders) [inline JiveAuthenticatingHTTPProtocolDemo/WKWebViewDemo/ViewController.m]: I don't think you should bring `JAHPAuthenticatingHTTPProtocolDelegate` into this at all. It isn't needed.

**Review by hborders — CHANGES_REQUESTED** [review]
Very minor request.

File: JiveAuthenticatingHTTPProtocolDemo/WKWebViewDemo/ViewController.m (hunk 1 above)
Comment (dodikk) [reply JiveAuthenticatingHTTPProtocolDemo/WKWebViewDemo/ViewController.m]: Deleted the mentioned `#pragma mark`

**Comment by dodikk** [comment]
Will be merged as part of a more recent PR https://github.com/jivesoftware/JiveAuthenticatingHTTPProtocol/pull/11

**Pull request closed without merging**